backend/app/controllers/uoas.js

```diff
@@ -533,3 +442,126 @@ module.exports = {
+                        .on(Product.projectId.equals(Project.id))
+                )
+                .where(
+                    Product.id.equals(productId)
+                )
+        );
+
+        if (_.first(project)) {
+            // If project is not active, delete UAO without any issues
+            if (_.first(project).status === 0) {
+
+                yield thunkQuery(
+                    'UPDATE "ProductUOA"' +
+                    'SET "isDeleted" = (to_timestamp(' + Date.now() +
+                    '/ 1000.0)) WHERE "UOAid" = ' + UOAId +
+                    'AND "productId" = ' + productId
+                );
+
+                // Soft delete the task with that UAO ID
+                yield thunkQuery(
+                    'UPDATE "Tasks"' +
+                    'SET "isDeleted" = (to_timestamp(' + Date.now() +
+                    '/ 1000.0)) WHERE "productId" = ' + productId +
+                    'AND "uoaId" = ' + UOAId
+                );
+
+                if (deleteOption === 1) {
+                    yield thunkQuery(
+                        'UPDATE "UnitOfAnalysis"' +
+                        'SET "isDeleted" = (to_timestamp(' + Date.now() +
+                        '/ 1000.0)) WHERE "id" = ' + UOAId
+                    );
+                }
+            } else { // Project is active and we have to do other checks.
+
+                // check if there are any tasks assigned
+                var tasks = yield thunkQuery(
+                    'SELECT "Tasks".* ' +
+                    'FROM "Tasks" ' +
+                    'WHERE "Tasks"."uoaId" = ' + UOAId +
+                    'AND "Tasks"."productId" = ' + productId
+                );
+
+                if (!_.first(tasks)) {
+                    // Soft delete the UOA from the Product UAO Table
+                    yield thunkQuery(
+                        'UPDATE "ProductUOA"' +
+                        'SET "isDeleted" = (to_timestamp(' + Date.now() +
+                        '/ 1000.0)) WHERE "UOAid" = ' + UOAId +
+                        'AND "productId" = ' + productId
+                    );
+
+                    if (deleteOption === 1) {
+                        // Soft delete from the UOA table
+                        yield thunkQuery(
+                            'UPDATE "UnitOfAnalysis"' +
+                            'SET "isDeleted" = (to_timestamp(' + Date.now() +
+                            '/ 1000.0)) WHERE "id" = ' + UOAId
+                        );
+                    }
+
+                } else {
+                    const modifiedTasksList = yield * common.getCompletenessForTask(req, tasks);
+
+                    if (_.first(modifiedTasksList).isComplete === true) {
+                        throw new HttpError(403, 'Cannot delete UOA of already completed task');
+                    } else {
+                        // TODO: Add check to make sure there aren't answered questions for survey Blocked by SER-160
```

Comment:
We need to check here that there are no answers to a survey question before deleting the Subject. Blocked by SurveyService. Tickers for those are:
https://jira.amida-tech.com/browse/SER-160
https://jira.amida-tech.com/browse/INBA-679

Reply:
@Perry5 either `/assessment-answers/{id/status` does what you want or there is a bug on the front end, because the client uses that endpoint to do this exact check

Reply:
Since we'll be working on this in a separate branch/ticket. Is this good to merge?